From f4b011bd2ad74e87f8610273f2929555869ab9a3 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 20 Jun 2023 15:42:50 +0200
Subject: [PATCH 0369/1141] soc: st: allow building STM32 HDP as module

Add the relevant declaration to build the driver as module.
Update the copyright date and email address.
Use 'tristate' in Kconfig.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I41e1c93d5480abe067893a546d8d0dc9576f975d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/313882
Tested-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Domain-Review: Eric FOURMONT <eric.fourmont-ext@st.com>
---
 drivers/soc/st/Kconfig     | 2 +-
 drivers/soc/st/stm32_hdp.c | 8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

--- a/drivers/soc/st/Kconfig
+++ b/drivers/soc/st/Kconfig
@@ -1,7 +1,7 @@
 if ARCH_STM32
 
 config STM32_HDP
-	bool "STMicroelectronics STM32 Hardware Debug Port (HDP) pin control"
+	tristate "STMicroelectronics STM32 Hardware Debug Port (HDP) pin control"
 	depends on MACH_STM32MP157 || MACH_STM32MP13 || MACH_STM32MP25 || ARM64
 	default n
 	help
--- a/drivers/soc/st/stm32_hdp.c
+++ b/drivers/soc/st/stm32_hdp.c
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright (C) STMicroelectronics 2018 - All Rights Reserved
- * Author: Christophe Roullier <christophe.roullier@st.com>
+ * Copyright (C) STMicroelectronics 2018-2023 - All Rights Reserved
+ * Author: Christophe Roullier <christophe.roullier@foss.st.com>
  * for STMicroelectronics.
  */
 
@@ -240,3 +240,7 @@ static struct platform_driver hdp_driver
 };
 
 module_platform_driver(hdp_driver);
+
+MODULE_AUTHOR("Christophe Roullier <christophe.roullier@foss.st.com>");
+MODULE_DESCRIPTION("STM32 Hardware Debug Port (HDP) pin control");
+MODULE_LICENSE("GPL");
